src/project_root.rs

```diff
@@ -1,7 +1,18 @@
 use glob::glob;
 use std::path::PathBuf;
+use std::collections::HashSet;
+use std::env;
 
-pub fn find_project_root(roots: &[String], path: &str) -> String {
+pub fn find_project_root(language: &str, markers: &[String], path: &str) -> String {
+    let vars = gather_env_roots(language);
+    if vars.is_empty() {
+    	roots_by_marker(markers, path)
+    } else {
+        roots_by_env(&vars, path)
```

Comment:
It worth running `cargo fmt` to make formatting consistent with the rest of the codebase.